### ISSUE TITLE
Raster upload data support ISO8601 timestamp and timezone information is interpreted incorrectly

### ISSUE THREAD

**Issue by lexvand** [nens/lizard-management-client#869]
The upload field doesn't recognize/translate '2021-06-24T050000Z', while this is closest to the ISO8601 standard a filename supports (leaving out colons). It does support without hyphens. Please support both formats.

Addition: Timezone information is not interpreted correctly as well. The `Z` indicates it is UTC. Another example could be `2021-06-24T050000+01:00`, which is in CET (UTC +1 hour, our winter time). UTC should also be the default when no timezone information is supplied.

**Requirements:**
- [x] file name with timestamp in this format: `YYYY-MM-DDTHHmmssZ` should be supported.
- [x] Timezone information is not interpreted correctly. Although the Z is included in the timestamp, it is not translated to UTC time when sending the data to backend.
- [x] When Z is not included in the timestamp then UTC should be used as default.
- [x] Make it clear which timezone the date time is displayed in the input field (UTC)
- [x] Support for other timezone formats (for example: `2021-06-24T050000+01:00`)? --> Yes, support --> convert to UTC time
- [x] When a timestamp is not present in the filename or the timestamp is in a wrong format then the date time input field should be empty instead of being pre-filled with the current date time
- [x] Add validation to the timestamp field as empty timestamp is not allowed
- [x] In the text "Drag and drop .tiff files here or ...", .tiff should be .tif 

**How to test:**
1. Go to https://nxt3.staging.lizard.net/management/#/data_management/rasters/sources and create or select a **temporal** raster source to go to the raster source form 
2. In the data field within the raster source form, either select browse to select a .tif file to upload or drag/drop the file to the upload section (make sure that the filename includes a timestamp in it)
3. Check in the date time input if the time is displayed correctly and is clear enough in which timezone
4. Click on Save and check in the network tab if a POST request is sent to the `data` endpoint with correct date time information

**Comment by hoanphungt** [comment]
@lexvand 
according to this wiki page: https://en.wikipedia.org/wiki/ISO_8601, this timestamp format `2021-06-24T050000Z` is incorrect.

ISO 8601 supports date and time in these formats:

1. `2021-06-28T11:22:40+00:00`
2. `2021-06-28T11:22:40Z`
3. `20210628T112240Z`

Do you mean that you want me to make an exception for the raster upload filename so it accepts this format `2021-06-24T050000Z` without colons?

Your second point about timezone is correct. The raster upload field can only translate to UTC time at the moment. So it does not matter if you include Z or `+01:00` or nothing, it will always be in UTC. Do you want me to take into account of timezone information too?

**Comment by lexvand** [comment]
If it is not too much work it would be nice to also support the format including hyphens and excluding colons, because that is also supported by our FTP importer.

The main issue with the timezone is that it is transposed from local time (input field) to UTC (API request). Meaning that filename `20210224T050000Z.tif` is interpreted as `24-2-2021 05:00` but posted with timestamp `"2021-06-24T04:00:00Z"`. By not including the `Z` in the interpretation it becomes unclear to users. We should be very explicit on this. If possible we should support timezone information. The explanation box should describe the timezone handling.

(And change ".tiff" to ".tif". The standard notation is either .tif or .geotiff.)

**Comment by hoanphungt** [comment]
@lexvand @RonaldWillemsen 
I updated the ticket with some requirements and steps to test it. Do you think that it is clear enough and if anything is missing/misleading?

**Comment by hoanphungt** [comment]
@lexvand 
I changed the estimate for this issue as 0.25 is too little for the number of tasks that I listed in the description.

**Comment by hoanphungt** [comment]
@Derryrover 
Hi Tom,

Could you review: https://github.com/nens/lizard-management-client/pull/871/?

Kr, Hoan

**Comment by Derryrover** [comment]
@hoanphungt 

Hi Hoan, 
I reviewed the code and tested the part where the dates are parsed.
I concluded the code looks good
I concluded all works correct to the specs in the description.
There is one potential problem:

If I upload a filename hoogte-1Z_2021-07-02T153350+01:00.tiff then the time UTC time prefilled will be 2021-07-02T153350
Maybe this is actually bad, because this is actually not what this file specifies and likely the user will miss this.
Maybe in the case of a timezone spec in the filename:
- the time should be corrected to utc (so timezone spec substracted or added)
- the time should stay empty (user has to fill it out manually)
- or we leave it as it is
Maybe ask Lex.

Further I have some trouble uploading raster tiff data on the backend (also on staging).
I can therefore not test if the uploading actually works.
All the uploaded files keep on processing.
Is this something we need to chase?

kr Tom

**Comment by hoanphungt** [comment]
@Derryrover 
Thanks for your input!

Regarding filename with +01:00, I actually discussed it with Lex and we agree that the field should be left empty in this case. I will improve this part.

Regarding the problem with uploading file in backend, I think at the time of your uploading, there is another big task in front of the queue and that's why your upload is not processed. It can happen from time to time. I just tried it again and it is done very quickly.
